### PR TITLE
more complex jsonAPI modeling

### PR DESCRIPTION
**Pull request by greydnls**

### CODE REVIEW

File: src/JsonApi/CompoundDocument.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function __construct(JsonApiResource $resource, array $includedData)
+    {
+        $this->resource = $resource;
+        $this->included = $includedData;
+    }
+
+    public function getRelatedResource(Relation $relation): ?JsonApiResource
+    {
+        $relationship = array_pop(array_filter($this->included, function (JsonApiResource $included) use ($relation) {
```
Comment (ieatkillerbees) [inline src/JsonApi/CompoundDocument.php]: This is a little hard to read and isn't PSR-2 ✌️

File: src/JsonApi/CompoundDocument.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        if ($relationship instanceof JsonApiResource) {
+            return $relationship;
+        }
+
+        throw new \InvalidArgumentException('Invalid Related Resource Requested');
+    }
+
+    public function hasRelatedResource(Relation $relation): bool
+    {
+        return array_pop(array_filter($this->included, function (JsonApiResource $included) use ($relation) {
```
Comment (ieatkillerbees) [inline src/JsonApi/CompoundDocument.php]: Samesies

**Review by ieatkillerbees — APPROVED** [review]
Non-blocking comments.

**Comment by ieatkillerbees** [comment]
👩‍🍳

File: src/JsonApi/CompoundDocument.php (hunk 1 above)
Comment (greydnls) [reply src/JsonApi/CompoundDocument.php]: 3psr5u

File: src/JsonApi/CompoundDocument.php (hunk 2 above)
Comment (greydnls) [reply src/JsonApi/CompoundDocument.php]: psr ur butt

**Pull request merged**